### PR TITLE
Remove es1 connections, library, container & rename es6 names

### PR DESCRIPTION
**Pull request by hmstepanek**
- [x]  Remove unused ELASTICSEARCH_HOST & _AWS* settings
- [x]  Remove unused get_client for es1
- [x]  Remove elasticsearchold from docker-compose
- [x]  Remove elasticsearch1 as a dependency 
- [x]  Rename request.es6 to request.es
- [x]  Rename es6_client fixture & search.get_es6_client
- [x]  Replace new_es6_index w/ new_index
- [x]  Remove es filter syntax for es1

This is a partial fix for https://github.com/hypothesis/product-backlog/issues/608.

The change to the docker-compose is simply to remove the old elasticsearch container. This can be achieved without re-running docker-compose and destroying the data in the Postgres container by just killing the old elasticsearch container:
```
docker kill h_elasticsearchold_1
```

### CODE REVIEW

**Comment by robertknight** [comment]
The changes in this PR look great, except that the tests are failing on Travis and would fail on Jenkins because they are still trying to start Elasticsearch 1 and 6 using the previous port assignments. I've marked the PR as WIP for that reason.

Testing this out locally I ran into the same issue but was able to resolve it with:

```bash
git checkout master
docker-compose stop
git checkout remove-request.es
docker-compose up
```

One way to resolve this I think would be to leave the port change out of this PR and leave it for a follow-up one. That will also give you a chance to alert the other devs about the need to do the above dance (if you know how it can be done in fewer commands, let me know!).

File: h/search/__init__.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
     'TagsAggregation',
     'UsersAggregation',
     'get_client',
-    'get_es6_client',
+    'get_client',
```
Comment (robertknight) [inline h/search/__init__.py]: `get_client` is now listed twice here.

**Comment by hmstepanek** [comment]
@robertknight I've updated the description with instruction on how to run with the new docker-compose. Let me know if they don't make sense!

**Comment by robertknight** [comment]
> Steps to run with the docker changes in an old env that has the old docker-compose still checked out:

`docker-compose down` will remove all the data in the Postgres container, at least assuming the developer created/ran the container in the first place using `docker-compose up` and didn't do anything to put the data in an external volume. That would probably be surprising and a nuisance for other developers.

I see that `docker-compose up` supports a `--remove-orphans` flag which I expected to remove the elasticsearchold container after running `docker-compose stop` and switching branches. It didn't do that though.

File: h/search/__init__.py [hunk 2]
```diff
@@ -38,24 +35,11 @@ def includeme(config):
 
     connect(hosts=[settings['es.url']], **kwargs)
 
-    # Connection to old (ES1.5) follows
-    settings.setdefault('es.host', 'http://localhost:9200')
-    settings.setdefault('es.index', 'hypothesis')
```
Comment (robertknight) [inline h/search/__init__.py]: Remove the default initialization of `es.index` here breaks the `hypothesis init` command.

Comment (hmstepanek) [reply h/search/__init__.py]: Oops that shouldn't have been removed. I fix.

**Comment by hmstepanek** [comment]
docker-compose stop as I understand it doesn’t remove the containers but for want of a better term, pauses them. I think I’m this particular case since we no longer are changing the port but rather just removing the old es container we should prob just run docker-compose kill on the old container.

**Comment by robertknight** [comment]
> docker-compose stop as I understand it doesn’t remove the containers but for want of a better term, pauses them.

Yes, this is my understanding too. It basically kills the root process in the container by sending it a signal but keeps the data volume & other resources that make up the container.

**Pull request merged**